### DISCUSSION TITLE
Problems with Simu5G

### DISCUSSION THREAD

**Discussion by MSklodowski** [omnetpp/omnetpp#1255] [Q&A]
Dear @rhornig , @giovanninardini

After many attempts, I still couldn't solve the problem with installing the Simu5G library. I've tried everything:
- Installing on different versions of Ubuntu and Omnet++
- using the opp_env method
- installing a virtual machine with Simu5G

They all led to the same thing: an error related to a segmentation fault. (picture below)

![image](https://github.com/user-attachments/assets/0f28400c-2e62-4fac-96ab-94d3d9a70925)

Below is a diagram of how I installed it all:
1. I installed Ubuntu version 20.04.6 LTS via a Bootable USB stick on a dual boot next to Windows 11. For that i used Rufus program.
2. I installed Omnet++ version 6.0.1 according to the instructions in this video:  https://www.youtube.com/watch?v=qKucVnIz3gU
3. I installed the INET library in version 4.4.2 via Omnet++ IDE using the "Help -> Install Simulation Models" option and additionally rebuilt it using CTRL + B
4. I downloaded the Simu5G library in version 1.2.1 from the repository (https://github.com/Unipisa/Simu5G/releases), unpacked it in workspace and imported it according to the instructions for OMNET++ IDE from the official website. I also converted it by CTRL + B

The above error pops up when I run the "MultiMecHost" simulation. The only case in which this simulation works correctly is when I install the libraries in the latest versions. However, Simu5G version 1.2.1 is required for my project.

What should I do in this situation? Thank you in advance for your help.

**Comment by rhornig** [comment]
This is NOT an installation related problem. This is a crash caused by a bug in Simu5G which was most likely fixed in Simu5G 1.2.2. So the solution is:

use Simu5G 1.2.2

> However, Simu5G version 1.2.1 is required for my project.

Why? As the 1.2.2 is just a patch release, there is no reason why it would not work for you instead on 1.2.1

**Comment by MSklodowski** [comment]
The problem is that my project requires a file swap in the src folder in the simu5G library. (an additional class was included there) This was prepared for version 1.2.1. I tried to do this for version 1.2.2 but the program did not work properly.

**Reply by rhornig** [reply]
Then, you should port your changed file to 1.2.2 (i.e. check what has changed between 1.2.1 and 1.2.2 in that file).

in short. create a DIFF file comparing the given file in 1.2.1 and in your project and then try to apply that DIFF file on top of 1.2.2

That's why it's never recommended to overwrite files in a project instead of applying DIFF files.
